packages/ui-library/src/elements/tag-input/index.js

```diff
@@ -38,16 +39,13 @@ const Tag = ( { tag, index, disabled = false, onRemoveTag, screenReaderRemoveTag
 
 	return (
 		// eslint-disable-next-line jsx-a11y/no-static-element-interactions
-		<div onKeyDown={ handleKeyDown } className="yst-tag-input__tag" { ...props }>
+		<Badge onKeyDown={ handleKeyDown } { ...props }>
 			<span>{ tag }</span>
-			<button
-				className="yst-tag-input__remove-tag"
-				onClick={ handleClick }
-			>
+			<button onClick={ handleClick }>
 				<span className="yst-sr-only">{ screenReaderRemoveTag }</span>
-				<XIcon className="yst-h-4 yst-w-4" />
+				<XIcon className="yst-h-3 yst-w-3 yst-opacity-50" />
```

Comment:
Interesting solution to get the color! But we might need some more control here.
It doesn't look quite right. For example, the hover and round effect are missing:
![image](https://user-images.githubusercontent.com/35524806/204850049-12f00685-b7e4-4c86-b0d5-df7f295ad463.png)